File: mass-core/src/main/scala/mass/model/job/RunStatus.java
package mass.model.job;

import com.fasterxml.jackson.annotation.JsonValue;
import fusion.json.CborSerializable;
import helloscala.common.data.IntEnum;

import java.util.Arrays;
import java.util.Optional;

public enum RunStatus implements IntEnum, CborSerializable {
    JOB_NORMAL(0, "NORMAL"),
    JOB_ENABLE(1, "ENABLE"),
    JOB_RUNNING(100, "RUNNING"),
    JOB_OK(200, "OK"),
    JOB_FAILURE(500, "FAILURE"),
    ;

    @JsonValue
    private Integer value;
    private String name;

    RunStatus(Integer value, String name) {
        this.value = value;
        this.name = name;
    }

    public Integer getValue() {
        return value;
    }

    public String getName() {
        return name;
    }

    public static Optional<RunStatus> optionalFromValue(int value) {
        return Arrays.stream(values()).filter(v -> v.value.equals(value)).findFirst();
    }

    public static RunStatus fromValue(int value) {
        return optionalFromValue(value).orElseThrow();
    }
}

File: mass-core/src/main/scala/mass/model/job/RunStatus.scala
//package mass.model.job
//
//import helloscala.common.util.{ EnumTrait, EnumTraitCompanion }
//
//sealed abstract class RunStatus(override val companion: EnumTraitCompanion, override protected val value: Int)
//    extends EnumTrait
//
//object RunStatus extends EnumTraitCompanion {
//  self =>
//  override type Value = RunStatus
//
//  case object JOB_NORMAL extends RunStatus(self, 0)
//  case object JOB_ENABLE extends RunStatus(self, 1)
//  case object JOB_RUNNING extends RunStatus(self, 100)
//  case object JOB_OK extends RunStatus(self, 200)
//  case object JOB_FAILURE extends RunStatus(self, 500)
//}
